modules/flowable-rest/src/main/java/org/flowable/rest/service/api/history/HistoricProcessInstanceIdentityLinkCollectionResource.java

```diff
@@ -53,9 +54,14 @@
             @ApiResponse(code = 404, message = "Indicates the process instance could not be found..") })
     @GetMapping(value = "/history/historic-process-instances/{processInstanceId}/identitylinks", produces = "application/json")
     public List<HistoricIdentityLinkResponse> getProcessIdentityLinks(@ApiParam(name = "processInstanceId") @PathVariable String processInstanceId, HttpServletRequest request) {
-        HistoricProcessInstance processInstance = getHistoricProcessInstanceFromRequest(processInstanceId);
-        
-        List<HistoricIdentityLink> identityLinks = historyService.getHistoricIdentityLinksForProcessInstance(processInstance.getId());
+        HistoricProcessInstance processInstance = historyService.createHistoricProcessInstanceQuery().processInstanceId(processInstanceId).singleResult();
+        if (processInstance == null) {
+            throw new FlowableObjectNotFoundException("Could not find a process instance with id '" + processInstanceId + "'.", HistoricProcessInstance.class);
+        }
```

Comment:
this bit can also be moved to `GetHistoricIdentityLinksForTaskCmd#getLinksForProcessInstance` which is called in the end by the `getHistoricIdentityLinksForProcessInstance` method

Reply:
also included in the cmd